packages/SwingSet/src/kernel/kernel.js

```diff
@@ -684,12 +686,40 @@ export default function buildKernel(
     return harden(policyInput);
   }
 
+  async function processStartVat(message) {
+    postAbortActions.discardFailedDelivery = true;
+    const { type, vatID } = message;
+    // console.log(`-- processStartVat(${vatID})`);
+    insistVatID(vatID);
+    // eslint-disable-next-line no-use-before-define
+    if (!vatWarehouse.lookup(vatID)) {
+      // Note a bit of weirdness here: if a vat's module initialization or
+      // buildRootObject function throws an error, the crank in which it was
+      // executed will be aborted, which will leave the 'startVat' delivery on
+      // the run queue (since the removal of the request from the run queue will
+      // have been rolled back by the abort).  This means it will be delivered
+      // *again* on the next crank; that, however, will lead us here: the vat
+      // won't exist (because it was terminated) and the nugatory redundant
+      // 'startVat' will safely render as a no-op and go away.
+      /** @type { PolicyInput } */
+      const policyInput = harden(['create-vat', {}]);
+      return harden(policyInput);
+    }
+    const kd = harden([type]); // TODO(4381) add vatParameters here
```

Comment:
let's not have this depend on `type`, it makes it look like `processStartVat` is intended to handle a variety of types, whereas really it must only ever be `startVat`.

```js
const kd = harden(['startVat']); // TODO(4381) add vatParameters here
```

Reply:
I'm not sure I agree.  Doing it the way it's written (which a number of the other `processXXX` functions also do, BTW) insulates the code from not having to be dependent on the name of the event it is processing; it just uses whatever name that happens to be.  So that if we were to, say, strictly hypothetically, change `'startVat'` to `'start-vat'`, this code would not have to be altered to match.